Meter component v1

chivchila opened the pull request.

chivchila committed:
testing prototype

chivchila commented on MIGRATION.md:
```diff
@@ -0,0 +1,12 @@
+_Migration guides are very important! Always include one for major releases. To create a codeblock that has diff highligting, use three backticks followed by the word diff_
```
Delete as we don't have any migrations to worry about

chivchila commented on README.md:
```diff
@@ -0,0 +1,86 @@
+- [Contact](#contact)
+- [Licence](#licence)
+
+_Whatever usage instructions your component has. We've broken this down by Markup, JavaScript and Sass, but it depends how complex your component is._
```
Write on how to customize it. Delete javascript etc... - things we don't need

chivchila commented on README.md:
```diff
@@ -0,0 +1,86 @@
+
+_Though it's not practical to repeat every aspect of Origami modules convention for every component, **A LOT** of people get tripped up by silent mode, so this line (remember to change the o-meter to your component name) is useful if you have Sass:_
+
+As with all Origami components, o-meter has a [silent mode](http://origami.ft.com/docs/syntax/scss/#silent-styles). To use its compiled CSS (rather than using its mixins with your own Sass) set `$o-meter-is-silent : false;` in your Sass before you import the o-meter Sass.
```
include o-meter mixin, remove mentioning of silent mode and above (whole sass section)

chivchila commented on README.md:
```diff
@@ -0,0 +1,86 @@
+## Contributing
+If your component is particularly complicated (image sets fall into this category) then a contributing section or even a contributing.md might be useful.
+
+## Migration
```
can be deleted

chivchila commented on README.md:
```diff
@@ -0,0 +1,86 @@
+⚠ maintained | 2 | 2.0 | [migrate to v2](MIGRATION.md#migrating-from-v1-to-v2) |
+╳ deprecated | 1 | 1.0 | N/A |
+
+## Contact
```
add browser support section. Explain IE situation

chivchila commented on bower.json:
```diff
@@ -0,0 +1,15 @@
+  ],
+  "main": [
+    "main.scss",
+    "main.js"
```
delete main.js as we aren't using js

chivchila commented on demos/src/demo.js:
```diff
@@ -0,0 +1,6 @@
+/*global require*/
+import './../../main.js';
```
delete anything that mentions javascript

chivchila commented on demos/src/demo.mustache:
```diff
@@ -0,0 +1,17 @@
+<div class="remove_me_later">
+	<div data-o-component="o-meter"></div>
```
tidy up to show 3 demos

chivchila commented on demos/src/demo.scss:
```diff
@@ -0,0 +1,9 @@
+@import '../../main';
+@include oMeter;
+
+.example {
```
add style attribute to meter 
put variables inline <meter class="o-meter-background: red" >

chivchila commented on main.js:
```diff
@@ -0,0 +1,10 @@
+import Meter from './src/js/meter';
```
remove

chivchila commented on main.scss:
```diff
@@ -0,0 +1,94 @@
+/// Output All oMeter Features
+@mixin oMeter() {
+
+	meter {
```
o-meter

chivchila commented on main.scss:
```diff
@@ -0,0 +1,94 @@
+		}
+	}
+
+	.remove_me_later {
```
remove

chivchila commented on origami.json:
```diff
@@ -0,0 +1,45 @@
+{
+	"description": "component description",
```
add

chivchila commented on origami.json:
```diff
@@ -0,0 +1,45 @@
+{
+	"description": "component description",
+	"keywords": ["meter", "progress", "range", "franction", "gauge", "data"],
```
franction

chivchila commented on main.scss:
```diff
@@ -0,0 +1,94 @@
+	meter {
+		--o-meter-background-color: white;
+		--o-meter-optimum-color: green;
+		--o-meter-sub-optimum-color: orange;
```
copy hex from origami (use mandarine for orange, crimson and jade)

chivchila commented on main.scss:
```diff
@@ -0,0 +1,94 @@
+		display: block;
+	}
+
+	.o-meter__container {
```
o-meter__container rename into o-meter-container because it's not a child, and we are following BEM
also, o-meter-value

chivchila commented on test/helpers/fixtures.js:
```diff
@@ -0,0 +1,34 @@
+let sandboxEl;
```
delete

chivchila commented on test/oMeter.test.js:
```diff
@@ -0,0 +1,52 @@
+/* eslint-env mocha */
```
delete

chivchila commented on test/scss/_mixins.test.scss:
```diff
@@ -0,0 +1,19 @@
+@include describe('oMeter mixins') {
```
if we want to write sass tests - it's the best place, otherwise delete

chivchila committed:
addressed feedback

chivchila committed:
setting brand as internal for demos

chivchila committed:
Deleted pa11y build config. The reason is below:
https://github.com/Financial-Times/o-meter/issues/2

notlee commented on README.md:
```diff
@@ -0,0 +1,73 @@
+		--o-meter-background-color: hotpink;
+		--o-meter-optimum-color: deeppink;
+		--o-meter-sub-optimum-colo: pink;
+		--o-meter-sub-sub-optimum-color: red;" data-o-component="o-meter" min="0" max="100" value="{{meterValue}}">
```
I love that Firefox went with `sub-sub-optimum` and webkit went with `even-less-good` naming for their CSS selector naming 😂 Maybe we should name our CSS variables after the [HTML attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meter) for these thresholds instead? There is `low`, `high`, `optimum`, we could go with: `--o-meter-low-color`, `--o-meter-high-color`, `--o-meter-optimum-color`

chivchila replied:
agree, those namings are amusing, indeed. I will change it

notlee commented on README.md:
````diff
@@ -0,0 +1,73 @@
+```
+
+## Support
+Meter tag is currently supported by Chrome, Safari, Firefox, Edge browsers. It is not supported by IE.
````
```suggestion
The `meter` tag is currently supported by Chrome, Safari, Firefox, Edge browsers. It is not supported by IE.
```

notlee commented on README.md:
```diff
@@ -0,0 +1,73 @@
+
+## Support
+Meter tag is currently supported by Chrome, Safari, Firefox, Edge browsers. It is not supported by IE.
+If needs to be used on IE, please use a fallback - include the value in the meter tag, for example:
```
```suggestion
If `o-meter` needs to be used on IE, please use a fallback - include the value in the meter tag, for example:
```

chivchila replied:
thanks!

notlee commented on main.scss:
```diff
@@ -0,0 +1,103 @@
+
+	.o-meter-value {
+		position: relative;
+		padding: 0 5px;
```
It would be great to use [o-spacing](https://registry.origami.ft.com/components/o-spacing@2.0.0/readme?brand=master) to set the padding. It helps us keep our margins and paddings a little more consistent.

chivchila replied:
good idea!

notlee commented on main.scss:
```diff
@@ -0,0 +1,103 @@
+		padding: 0 5px;
+		background: #{oColorsByName('black')};
+		color: #{oColorsByName('white')};
+		line-height: 24px;
```
We should include `o-typography` and set the font size / line height that way using o-typography mixins.

chivchila replied:
great! thanks for the suggestion!

notlee reviewed (commented):
This is looking great 😄

chivchila committed:
Update README.md

Co-Authored-By: Lee Moody <notlee@users.noreply.github.com>

chivchila committed:
Update README.md

Co-Authored-By: Lee Moody <notlee@users.noreply.github.com>

chivchila committed:
changing namings and adding mixins

JakeChampion commented on main.scss:
```diff
@@ -0,0 +1,104 @@
+		padding: 0 oSpacingByName('s1');
+		background: #{oColorsByName('black')};
+		color: #{oColorsByName('white')};
+		@include oTypographySans($scale: -1, $line-height: 24px);
```
Our linting rules require a mixin to be used before other CSS properties are declared. We have this linting rule because it helps us be sure that the mixin is not going to properties are not going to overwrite the any of the other CSS properties in the rule block.

chivchila replied:
thank you! got it!

chivchila committed:
fixing mixin declaration error

JakeChampion commented:
This is looking really good! 😄

notlee commented:
Hey, I've just taken another look to see if we're ready to release and spotted a couple of things!

- I think high and low are the wrong way around? `low` is bad, `high` is good, `optimum` is best
- The second demo puts the percentage twice, once for the arrow and once for when meter is not supported. Will it be read our to screen reader users twice? We should test that and maybe add the `aria-hidden` hidden attribute to one.
- The second demo sets an inline style `style="left: {{meterValue}}%"`. That's great but it might be even easier and more flexible to set a CSS variable `--o-meter-value: {{meterValue}}` instead?

chivchila committed:
Addressing 4 issues

1) Setting the right color for less optinal value
2) Hiding an optional value box where browsers don't support meter tag
3) Adding aria-label for demo examples
4) Addressing value and percentage difference in an example

chivchila committed:
add disable no-vendor-prefixes in the right place

chivchila commented:
Thank you, Lee! I have changed the colors and added aria-hidden. As we discussed we won't set  css variable --o-meter-value as the meter values will not really be repeating - value and percentage that we are setting on the meter are two different things, even though their values can match. I have renamed a value in an optional span to be called 'percentValue' in demos

notlee commented on demos/src/demo.1.mustache:
```diff
@@ -0,0 +1,3 @@
+<meter class="o-meter" aria-label="demo meter {{meterValue}}" data-o-component="o-meter" min="0" max="100" value="{{meterValue}}">
```
We just need to update the markup in the README with the aria-label. Otherwise this is great and ready to go :D

chivchila replied:
ah yes! of course! thank you!

chivchila committed:
add aria-label to readme file

chivchila committed:
change component status to maintained
